sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -324,7 +324,8 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
       DescribeTableCommand(
         visitTableIdentifier(ctx.tableIdentifier),
         partitionSpec,
-        ctx.EXTENDED != null || ctx.FORMATTED != null)
+        ctx.EXTENDED != null,
+        ctx.FORMATTED != null)
```

Comment:
For `DESC t`, we should change it back to be compatible with before, for `DESC EXTENDED`, it should be same as `DESC FORMATTED`, so we don't need to change it.